tools/Vrops.py

```diff
@@ -149,368 +124,120 @@ def get_resources(self, target, token, uuids, resourcekinds):
-        else:
-            logger.error(f'Return code: {response.status_code} != 200 for {key} : {response.text}')
-            return False
-
-    # this is for a single query of a property and returns the value only
-    def get_property(target, token, uuid, key, collector):
-        logger.debug('>------------------------------------- get_property')
-        logger.debug(f'key      : {key}')
-        logger.debug(f'target   : {target}')
-        logger.debug(f'collector: {collector}')
-
-        url = "https://" + target + "/suite-api/api/resources/" + uuid + "/properties"
-        headers = {
-            'Content-Type': "application/json",
-            'Accept': "application/json",
-            'Authorization': "vRealizeOpsToken " + token
-        }
-        disable_warnings(exceptions.InsecureRequestWarning)
-        try:
-            response = requests.get(url,
-                                    verify=False,
-                                    headers=headers)
-        except Exception as e:
-            logger.error(f'Problem getting stats error for {key} - Error: {e}')
-            return False
-
-        if response.status_code == 200:
-            try:
-                for propkey in response.json()["property"]:
-                    if propkey["name"] is not None and propkey["name"] == key:
-                        logger.debug('<--------------------------------------------------')
-                        return propkey["value"]
-            except IndexError as e:
-                logger.error(f'Problem getting property error for {key} - Error {e}')
-        else:
-            logger.error(f'Return code: {response.status_code} != 200 for {key} : {response.text}')
-            return False
-
-    # if we expect a number without special characters
-    def get_latest_number_properties_multiple(target, token, uuids, propkey, collector):
-        if not isinstance(uuids, list):
-            logger.error('Error in get project_ids: uuids must be a list with multiple entries')
-            return False
-
-        logger.debug('>------------ get_latest_number_properties_multiple')
-        logger.debug(f'key      : {propkey}')
-        logger.debug(f'target   : {target}')
-        logger.debug(f'collector: {collector}')
+    def _get_chunk(q, uuid_list, url, headers, keys, target, kind, collector, chunk_iteration):
+        logger.debug(f'chunk: {chunk_iteration}')
 
-        return_list = list()
-        url = "https://" + target + "/suite-api/api/resources/properties/latest/query"
-        headers = {
-            'Content-Type': "application/json",
-            'Accept': "application/json",
-            'Authorization': "vRealizeOpsToken " + token
-        }
         payload = {
-            "resourceIds": uuids,
-            "propertyKeys": [propkey]
+            "resourceId": uuid_list,
+            "statKey": keys,
+            "PageSize": 500000
```

Comment:
is this the maximum? Or a random picked number?
How many entries are we getting back currently?

Reply:
This is just a guess. I will do a test to ensure the right value here.